fiddler/tests/test_fiddler.py

```diff
@@ -1,15 +1,31 @@
 from typing import Any, Callable, Dict
 
+import pytest
+
 from datadog_checks.base import AgentCheck
 from datadog_checks.base.stubs.aggregator import AggregatorStub
 from datadog_checks.dev.utils import get_metadata_metrics
 from datadog_checks.fiddler import FiddlerCheck
 
+from .conftest import FIDDLER_API_KEY
+
 
+# Minimal test
 def test_check(dd_run_check, aggregator, instance):
     # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
     check = FiddlerCheck('fiddler', {}, [instance])
+    # Prevent the integration from failing
+    with pytest.raises(Exception) as e:
+        dd_run_check(check)
+    assert 'Authorization Required' in str(e.value)
+
+
+@pytest.mark.skipif(not FIDDLER_API_KEY, reason="Local testing only")
+def test_metric_collection(dd_run_check, aggregator, e2e_instance):
+    # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
+    check = FiddlerCheck('fiddler', {}, [e2e_instance])
```

Comment:
This test can be run locally when the environment variable is set

Reply:
Sure.